rig-core/src/providers/openai.rs

```diff
@@ -205,12 +208,60 @@ enum ApiResponse<T> {
 // ================================================================
 // OpenAI Embedding API
 // ================================================================
-/// `text-embedding-3-large` embedding model
-pub const TEXT_EMBEDDING_3_LARGE: &str = "text-embedding-3-large";
-/// `text-embedding-3-small` embedding model
-pub const TEXT_EMBEDDING_3_SMALL: &str = "text-embedding-3-small";
-/// `text-embedding-ada-002` embedding model
-pub const TEXT_EMBEDDING_ADA_002: &str = "text-embedding-ada-002";
+#[derive(Debug, Clone)]
+pub enum OpenAIEmbeddingModel {
+    TextEmbedding3Large,
+    TextEmbedding3Small,
+    TextEmbeddingAda002,
+}
```

Comment:
Keep as `const &str` and maybe match on string to get the embedding vector dimension?

The idea is to allow users to specify a model for which we don't have consts/enums for (e.g.: a model that was just release and that we did not have time to integrate yet)